security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -304,6 +305,24 @@ func (sc *SecretCache) rotate() {
 }
 
 func (sc *SecretCache) generateSecret(ctx context.Context, token, resourceName string, t time.Time) (*model.SecretItem, error) {
+	// If node agent works as ingress gateway agent, searches for kubernetes secret instead of sending
+	// CSR to CA.
+	if sc.fetcher.UseCaClient == false {
+		secretItem, exist := sc.fetcher.FindIngressGatewaySecret(resourceName)
+		if !exist {
+			return nil, fmt.Errorf("cannot find secret %s for ingress gateway", resourceName)
+		}
+
+		return &model.SecretItem{
+			CertificateChain: secretItem.CertificateChain,
+			PrivateKey:       secretItem.PrivateKey,
+			ResourceName:     resourceName,
+			Token:            token,
```

Comment:
in your case token isn't required, you may need to update ```getCredentialToken``` in sdsservice.go to remove that constraint in your case.